packages/block-editor/src/components/block-preview/index.js

```diff
@@ -2,209 +2,70 @@
-
-				// Hack: we need  to reset the scaled elements margins
-				previewElement.style.marginTop = '0';
-			} else {
-				const containerElementRect = containerElement.getBoundingClientRect();
-				scale = containerElementRect.width / viewportWidth;
-				setPreviewScale( scale );
-			}
-
-			setIsReady( true );
-			onReady( {
-				scale,
-				position: { x: offsetX, y: offsetY },
-				previewContainerRef: previewRef,
-
-				inlineStyles: getInlineStyles(
-					scale,
-					offsetX,
-					offsetY,
-					true,
-					viewportWidth
-				),
-			} );
-		}, scalingDelay );
-
-		// Cleanup
-		return () => {
-			if ( timerId ) {
-				window.clearTimeout( timerId );
-			}
-		};
-	}, [] );
-
-	if ( ! blocks || blocks.length === 0 ) {
-		return null;
-	}
-
-	const previewStyles = getInlineStyles(
-		previewScale,
-		x,
-		y,
-		isReady,
-		viewportWidth
-	);
-
-	return (
-		<div
-			ref={ previewRef }
-			className={ classnames(
-				'block-editor-block-preview__container editor-styles-wrapper',
-				{
-					'is-ready': isReady,
-				}
-			) }
-			aria-hidden
-		>
-			<Disabled
-				style={ previewStyles }
-				className="block-editor-block-preview__content"
-			>
-				<BlockList />
-			</Disabled>
-		</div>
-	);
-}
+import ScaledBlockPreview from './scaled';
+import AutoHeightBlockPreview from './auto';
 
+/**
+ * BlockPreview renders a preview of a block or array of blocks.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-preview/README.md
+ *
+ * @param {Array|Object} blocks A block instance (object) or an array of blocks to be previewed.
+ * @param {number} viewportWidth Width of the preview container in pixels. Controls at what size the blocks will be rendered inside the preview. Default: 700.
+ * @return {WPComponent} The component to be rendered.
+ */
 export function BlockPreview( {
 	blocks,
 	viewportWidth = 700,
 	padding,
-	settings,
+	autoHeight = false,
 	__experimentalOnReady = noop,
 	__experimentalScalingDelay = 100,
 } ) {
+	const settings = useSelect( ( select ) =>
+		select( 'core/block-editor' ).getSettings()
+	);
 	const renderedBlocks = useMemo( () => castArray( blocks ), [ blocks ] );
 	const [ recompute, triggerRecompute ] = useReducer(
 		( state ) => state + 1,
 		0
 	);
 	useLayoutEffect( triggerRecompute, [ blocks ] );
+	if ( ! blocks || blocks.length === 0 ) {
+		return null;
+	}
 	return (
 		<BlockEditorProvider value={ renderedBlocks } settings={ settings }>
 			{ /*
 			 * The key prop is used to force recomputing the preview
 			 * by remounting the component, ScaledBlockPreview is not meant to
 			 * be rerendered.
 			 */ }
-			<ScaledBlockPreview
-				key={ recompute }
-				blocks={ renderedBlocks }
-				viewportWidth={ viewportWidth }
-				padding={ padding }
-				onReady={ __experimentalOnReady }
-				scalingDelay={ __experimentalScalingDelay }
-			/>
+			{ autoHeight ? (
+				<AutoHeightBlockPreview viewportWidth={ viewportWidth } />
```

Comment:
does not autoHeight mode depend on the content?

Reply:
yes but that's provided by the BlockEditorProvider store.

Reply:
yes, but wondering if it should add the recompute key also to the  `< AutoHeightBlockPreview />` in order to force the rendering when it triggers a recomputation.

Reply:
Makes sense